src/blame.rs

```diff
@@ -0,0 +1,167 @@
+use std::marker;
+use {raw, Repository, Oid, signature, Signature};
+use util::{self, Binding};
+use std::path::Path;
+use std::ops::Range;
+
+/// Opaque structure to hold blame results.
+pub struct Blame<'repo> {
+    raw: *mut raw::git_blame,
+    _marker: marker::PhantomData<&'repo Repository>,
+}
+
+/// Structure that represents a blame hunk.
+pub struct BlameHunk<'blame> {
+    raw: *mut raw::git_blame_hunk,
+    _marker: marker::PhantomData<&'blame raw::git_blame>,
+}
+
+/// An iterator over the hunks in a blame.
+pub struct BlameIter<'blame> {
+    range: Range<usize>,
+    blame: &'blame Blame<'blame>,
+}
+
+impl<'repo> Blame<'repo> {
+
+    /// Gets the number of hunks that exist in the blame structure.
+    pub fn len(&self) -> usize {
+        unsafe { raw::git_blame_get_hunk_count(self.raw) as usize }
+    }
+
+    /// Gets the blame hunk at the given index.
+    pub fn get_index(&self, index: u32) -> Option<BlameHunk> {
+        unsafe {
+            let ptr = raw::git_blame_get_hunk_byindex(self.raw(), index);
+            if ptr.is_null() {
+                None
+            } else {
+                Some(BlameHunk::from_raw_const(ptr))
+            }
+        }
+    }
+
+    /// Gets the hunk that relates to the given line number in the newest
+    /// commit.
+    pub fn get_line(&self, lineno: u32) -> Option<BlameHunk> {
+        unsafe {
+            let ptr = raw::git_blame_get_hunk_byline(self.raw(), lineno);
+            if ptr.is_null() {
+                None
+            } else {
+                Some(BlameHunk::from_raw_const(ptr))
+            }
+        }
+    }
+
+    /// Returns an iterator over the hunks in this blame.
+    pub fn iter(&self) -> BlameIter {
+        BlameIter { range: 0..self.len(), blame: self }
+    }
+
+}
+
+impl<'blame> BlameHunk<'blame> {
+
+    unsafe fn from_raw_const(raw: *const raw::git_blame_hunk)
+                                 -> BlameHunk<'blame> {
+        BlameHunk {
+            raw: raw as *mut raw::git_blame_hunk,
+            _marker: marker::PhantomData,
+        }
+    }
+
+    /// Returns OID of the commit where this line was last changed.
+    pub fn id(&self) -> Oid {
```

Comment:
Perhaps this should be called `final_commit_id` to leave from for `orig_commit_id` to be exposed?